WP3/D3.8/report.tex

```diff
@@ -185,13 +185,94 @@ \section{Continuous integration achievements of OpenDreamKit projects}\label{pro
 
 \subsection{SageMath}
 
-% Julian should remark on his efforts here: In particular what problems
-% motivated his efforts, and how he thinks this will improve things in the
-% future.  Maybe make note here about producing Docker images as build
-% artifacts (TODO: confirm if GAP is doing this too), and further how
-% mybinder.org turns out to be a useful development tool in conjunction
-% with Dockerized builds.  Not sure how much to say about that here, or
-% leave it to the conclusion.
+The SageMath project had traditionally employed two distinct CI systems, the
+Buildbot and Sage's own development, the
+patchbot\footnote{\url{https://github.com/sagemath/sage-patchbot}}.
+
+\subsubsection{Buildbot and patchbot}
+Sage's Buildbot runs as a non-advisory CI system on a wide variety of
+platforms. It builds Sage and runs its test suite for every change that gets
+accepted into the mainline of the Sage code. Its main purpose is not to be an
+advisory tool for developers but to catch problems and incompatibilities of
```

Comment:
Of course, the way it is used currently, in combination with Volker, it does kind of serve an advisory role since he is basically manually merging changes, testing them, and the backing them out if they don't integrate.  But I think maybe that's a fine point we don't want to go into detail on here.

If I were him (and if we had a better advisory CI system) I would just merge changes directly into master, and in the relatively rare case that something breaks on one platform or another I would open a critical issue to fix it (assuming it's a blocker, on a platform we care about).

I don't necessarily think one way is "better" than the other, but his process seems much more labor intensive.  I can abide a broken build as long as it's quickly fixed.  He cannot abide ever having a broken build, it seems :|

Reply:
Yes, I was aware that it is technically advisory. For developers it does not feel like an advisory CI, however.
I agree that the current style seems relatively labor intensive though much more automated than what we used to do if I'm informed correctly.